amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueContainer.java

```diff
@@ -40,7 +62,64 @@ public static void putQueue(NamespaceName namespaceName, String queueName, AmqpQ
         });
     }
 
-    public static AmqpQueue getQueue(NamespaceName namespaceName, String queueName) {
+    /**
+     * Get or create queue.
+     *
+     * @param namespaceName namespace in pulsar
+     * @param queueName name of queue
+     * @param createIfMissing true to create the queue if not existed
+     *                        false to get the queue and return null if not existed
+     * @return the completableFuture of get result
+     */
+    public CompletableFuture<AmqpQueue> asyncGetQueue(NamespaceName namespaceName, String queueName,
+                                                      boolean createIfMissing) {
+        CompletableFuture<AmqpQueue> queueCompletableFuture = new CompletableFuture<>();
+        if (namespaceName == null || StringUtils.isEmpty(queueName)) {
+            log.error("Parameter error, namespaceName or queueName is empty.");
+            queueCompletableFuture.complete(null);
+            return queueCompletableFuture;
+        }
+        if (pulsarService.getState() != PulsarService.State.Started) {
+            log.error("Pulsar service not started.");
+            queueCompletableFuture.completeExceptionally(new PulsarServerException("PulsarService not start"));
+            return queueCompletableFuture;
+        }
+        Map<String, AmqpQueue> map = queueMap.getOrDefault(namespaceName, null);
+        if (map == null || map.getOrDefault(queueName, null) == null) {
```

Comment:
Same as the exchange container, we must ensure only create a queue for a queue name.

Reply:
Done.